stream-chat-android-client/src/main/java/io/getstream/chat/android/client/extensions/StringExtensions.kt

```diff
@@ -57,3 +64,135 @@ public fun String.cidToTypeAndId(): Pair<String, String> {
+    @FloatRange(from = 0.0, to = 1.0, fromInclusive = false) resizedHeightPercentage: Float,
+    resizeMode: StreamCdnResizeImageMode? = null,
+    cropMode: StreamCdnCropImageMode? = null,
+): String {
+    val logger = StreamLog.getLogger("Chat:resizedStreamCdnImageUrl")
+
+    val streamCdnImageDimensions = this.getStreamCdnHostedImageDimensions()
+
+    return if (streamCdnImageDimensions != null) {
+
+        val imageLinkUri = this.toUri()
+
+        if (imageLinkUri.wasImagePreviouslyResized()) {
+            logger.w {
+                "Image URL already contains resizing parameters. Please apply resizing parameters only to " +
+                    "original image URLs."
+            }
+
+            return this@createResizedStreamCdnImageUrl
+        }
+
+        val resizedWidth: Int = (streamCdnImageDimensions.originalWidth * resizedWidthPercentage).toInt()
+        val resizedHeight: Int = (streamCdnImageDimensions.originalHeight * resizedHeightPercentage).toInt()
+
+        val resizedImageUrl = imageLinkUri
+            .buildUpon()
+            .appendValueAsQueryParameterIfNotNull(key = "w", value = resizedWidth)
+            .appendValueAsQueryParameterIfNotNull(key = "h", value = resizedHeight)
+            .appendValueAsQueryParameterIfNotNull(key = "resize", value = resizeMode?.queryParameterName)
+            .appendValueAsQueryParameterIfNotNull(key = "crop", value = cropMode?.queryParameterName)
+            .build()
+            .toString()
+
+        logger.i {
+            "Resized Stream CDN hosted image URL: $resizedImageUrl"
+        }
+
+        resizedImageUrl
+    } else {
+        logger.w {
+            "Only images hosted by Stream's CDN containing original width and height query parameters " +
+                "can be resized"
+        }
+        this
+    }
+}
+
+/**
+ * Checks if a string contains resizing related query parameters.
+ *
+ * @return true if the URL contains resizing parameters, false otherwise.
+ */
+private fun Uri.wasImagePreviouslyResized(): Boolean =
+    queryParameterNames.run {
+        contains("w") || contains("h") || contains("resize") || contains("crop")
+    }
+
+/**
+ * A convenience method which evaluates if [value] is null or not and appends
+ * it with the accompanying parameter key name in the form of a query parameter.
+ *
+ * @param key Query parameter key.
+ * @param value Query parameter value.
+ */
+private fun Uri.Builder.appendValueAsQueryParameterIfNotNull(key: String, value: String?): Uri.Builder {
+    return if (value != null) {
+        this.appendQueryParameter(key, value)
+    } else {
+        this
+    }
+}
+
+/**
+ * A convenience method which evaluates if [value] is null or not and appends
+ * it with the accompanying parameter key name in the form of a query parameter.
+ *
+ * @param key Query parameter key.
+ * @param value Query parameter value.
+ */
+private fun Uri.Builder.appendValueAsQueryParameterIfNotNull(key: String, value: Int?): Uri.Builder {
```

Comment:
We could avoid overloading by using `?.toString()` instead of `.toString()`.

However the linter does not understand that this actually has a use and we would have to suppress the warning. Not an issue in itself, but I'm worried someone who doesn't know the reason behind this might see it in the future, think that it was a mistake and remove it. 

If that happens, when the value is null the `Uri.Buidlder` will append a query param as `querParam=null` instead of omitting it.